src/viam/robot/client.py

```diff
@@ -119,26 +130,38 @@ async def with_channel(cls, channel: Union[Channel, ViamChannel], options: Optio
         else:
             self._channel = channel.channel
             self._viam_channel = channel
+        self._connected = True
         self._client = RobotServiceStub(self._channel)
         self._manager = ResourceManager()
         self._lock = Lock()
         self._resource_names = []
         self._should_close_channel = False
+        self._options = options
         await self.refresh()
 
         if options.refresh_interval > 0:
             self._refresh_task = asyncio.create_task(
                 self._refresh_every(options.refresh_interval), name=f"{viam._TASK_PREFIX}-robot_refresh_metadata"
             )
 
+        if options.check_connected_time > 0 and options.reconnect_every > 0:
```

Comment:
@dannenberg @njooma this is a bit opinionated in that, if a user sets either of these to `0`, then they will continue to see the same behavior as before. Only when both are set will errors be immediately seen. It's based on the `Go` logic, but I'm happy to see this changed as appropriate.

Reply:
I changed the logic such that we always check for connection and log and at minimum log any errors. Then if enabled, also try and reconnect